python-avd/pyavd/_eos_designs/shared_utils/filtered_connected_endpoints.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING, Protocol
+
+from pyavd._eos_designs.schema import EosDesigns
+from pyavd._errors import AristaAvdError
+
+if TYPE_CHECKING:
+    from . import SharedUtilsProtocol
+
+
+class FilteredConnectedEndpointsMixin(Protocol):
```

Comment:
I think we can move the logic of both `FilteredConnectedEndpointsMixin` and `FilteredNetworkPortsMixin` to a single file as it was before.
@ClausHolbechArista